BAL/EmployeeBAL.cs

```diff
@@ -0,0 +1,148 @@
+using EmployeeManagementSystem.DAL;
+using EmployeeManagementSystem.Models;
+using EmployeeManagementSystem.Utils;
+namespace EmployeeManagementSystem.BAL;
+
+public class EmployeeBAL : IEmployeeBAL
+{
+    public readonly IEmployeeDAL employeeDal;
+    public readonly ILogger logger;
+
+    public EmployeeBAL(ILogger logger, IEmployeeDAL employeeDal)
+    {
+        this.employeeDal = employeeDal;
+        this.logger = logger;
+    }
+    public (List<EmployeeDetails>? employees, bool status) GetAllEmployees()
+    {
+        bool status;
+        List<EmployeeDetails> employees = new List<EmployeeDetails>();
+        try
+        {
+            (employees, status) = employeeDal.RetrieveAllEmployees();
+            if (!status && (employees == null || employees.Count == 0))
+            {
+                logger.LogError("No Employee(s) Found");
```

Comment:
Showing messages can be handled at consoler layer. Here you don't need to show messages. Also why do you log error where are no employees. It is just a info message - not error.